woocommerce/woocommerce-android
# Track unexpected date format

Part of: #10172

### Why
>On the MyStore screen, when displaying the revenue stats, we are trying to parse a date that is not in the expected format. From our first review of this issue, we think this could be a problem of either a race condition or a collision of the cached DB keys. More context here: peaMlT-c8-p2.

### Description
This PR aims to gather more information about the issue's root by tracking extra information to help us understand the possible causes.

### Testing instructions
TC1
1. Check that in a normal scenario, the `STATS_UNEXPECTED_FORMAT`  track event is not sent

TC2
1. Apply the patch and check that the `STATS_UNEXPECTED_FORMAT`  track event is sent

<details>
  <summary>Patch</summary>

```
Index: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/mystore/MyStoreStatsView.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/mystore/MyStoreStatsView.kt b/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/mystore/MyStoreStatsView.kt
--- a/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/mystore/MyStoreStatsView.kt	(revision be87c4c908cbab8b3167bdd6aa20482a7a8c46e6)
+++ b/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/mystore/MyStoreStatsView.kt	(date 1705614678078)
@@ -368,7 +368,10 @@
     private fun updateDateOnScrubbing(dateString: String, activeGranularity: StatsGranularity) {
         statsDateValue.text = when (activeGranularity) {
             StatsGranularity.DAYS -> dateUtils.getFriendlyDayHourString(dateString).orEmpty()
-            StatsGranularity.WEEKS -> dateUtils.getShortMonthDayString(dateString).orEmpty()
+            StatsGranularity.WEEKS -> {
+                val invalidDateFormat = "2014-01" 
+                dateUtils.getShortMonthDayString(invalidDateFormat).orEmpty()
+            }
             StatsGranularity.MONTHS -> dateUtils.getLongMonthDayString(dateString).orEmpty()
             StatsGranularity.YEARS -> dateUtils.getFriendlyLongMonthYear(dateString).orEmpty()
         }.also { result -> trackUnexpectedFormat(result, dateString) }

```

</details>
